### DISCUSSION TITLE
The `graphql-ws` subprotocol (aka the graphql-transport-ws package) has been deprecated

### DISCUSSION THREAD

**Discussion by DoctorJohn** [strawberry-graphql/strawberry#2072] [General]
The `graphql-ws` protocol [has been deprecated in March](https://github.com/apollographql/subscriptions-transport-ws/commit/ccf17cb24c9f38dec3443ca48527eab4c38eb9e8) and the corresponding [graphql-transport-ws repository](https://github.com/apollographql/subscriptions-transport-ws) has been archived by its maintainers. They recommend using the `graphql-transport-ws` subprotocol which lives in the [graphql-ws repository](https://github.com/enisdenjo/graphql-ws) and is actively maintained.

The question is **(for how long) do we want to keep supporting that protocol?**

Some random thoughts:

- The old protocols specification leaves a lot of room for interpretation and a few edge cases are not addressed at all.
- Removing the old protocol would simplify websocket subprotocol selection.
- It could also simplify the GraphiQL template, that is until we decide to add support for subscriptions via server sent events ;)
- The Apollo GraphQL docs [recommend the new protocol](https://www.apollographql.com/docs/react/data/subscriptions/#1-install-required-libraries).
- The Relay docs have [instructions for both protocols](https://relay.dev/docs/guided-tour/updating-data/graphql-subscriptions/) but they prefer the new one.
- Both the [old](https://www.npmjs.com/package/subscriptions-transport-ws) and the [new](https://www.npmjs.com/package/graphql-ws) npm packages still have about the same amount of weekly downloads

**Comment by patrick91** [comment]
> Both the [old](https://www.npmjs.com/package/subscriptions-transport-ws) and the [new](https://www.npmjs.com/package/graphql-ws) npm packages still have about the same amount of weekly downloads

Shall we wait until this changes quite a bit? I'd love to remove code, but I also want to allow people to update to latest versions of strawberry, if we remove the support for it then it will be harder.

I guess we can mark the old protocol as deprecated for now?

Another option might be to move the old protocol stuff to a new package, but I don't know if that's easy to do
